homeassistant/components/water_heater/__init__.py

```diff
@@ -288,42 +289,42 @@ def is_away_mode_on(self) -> bool | None:
         """Return true if away mode is on."""
         return self._attr_is_away_mode_on
 
-    def set_temperature(self, **kwargs):
+    def set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         raise NotImplementedError()
 
-    async def async_set_temperature(self, **kwargs):
+    async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         await self.hass.async_add_executor_job(
             ft.partial(self.set_temperature, **kwargs)
         )
 
-    def set_operation_mode(self, operation_mode):
+    def set_operation_mode(self, operation_mode: str) -> None:
         """Set new target operation mode."""
         raise NotImplementedError()
 
-    async def async_set_operation_mode(self, operation_mode):
+    async def async_set_operation_mode(self, operation_mode: str) -> None:
         """Set new target operation mode."""
         await self.hass.async_add_executor_job(self.set_operation_mode, operation_mode)
 
-    def turn_away_mode_on(self):
+    def turn_away_mode_on(self) -> None:
         """Turn away mode on."""
         raise NotImplementedError()
 
-    async def async_turn_away_mode_on(self):
+    async def async_turn_away_mode_on(self) -> None:
         """Turn away mode on."""
         await self.hass.async_add_executor_job(self.turn_away_mode_on)
 
-    def turn_away_mode_off(self):
+    def turn_away_mode_off(self) -> None:
         """Turn away mode off."""
         raise NotImplementedError()
 
-    async def async_turn_away_mode_off(self):
+    async def async_turn_away_mode_off(self) -> None:
         """Turn away mode off."""
         await self.hass.async_add_executor_job(self.turn_away_mode_off)
 
     @property
-    def min_temp(self):
+    def min_temp(self) -> float:
```

Comment:
Maybe this should allow `None`?
```suggestion
    def min_temp(self) -> float | None:
```

Reply:
I'm not sure how the frontend would handle this. What would it mean if the state and the max temp are present but min temp is None?

We currently do not allow this on the climate entity either for the same reasoning.

Reply:
I asked because that's it was defined as nullable in `melcloud`, so I guess that was a bug and fixing it is correct?

Reply:
It is probably an edge case in melcloud that is almost impossible to trigger, anyways, I think we should keep it as `float` now.